data/src/main/java/me/androidbox/data/local/agenda/TaskRepositoryImp.kt

```diff
@@ -0,0 +1,102 @@
+package me.androidbox.data.local.agenda
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import me.androidbox.data.local.dao.TaskDao
+import me.androidbox.data.local.entity.TaskSyncEntity
+import me.androidbox.data.mapper.toTask
+import me.androidbox.data.mapper.toTaskDto
+import me.androidbox.data.mapper.toTaskEntity
+import me.androidbox.data.remote.network.event.EventService
+import me.androidbox.data.remote.network.task.TaskService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.domain.agenda.model.Task
+import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.domain.constant.SyncAgendaType
+import me.androidbox.domain.task.repository.TaskRepository
+import javax.inject.Inject
+
+class TaskRepositoryImp @Inject constructor(
+    private val taskDao: TaskDao,
+    private val taskService: TaskService
+) : TaskRepository {
+    override suspend fun insertTask(task: Task): ResponseState<Unit> {
+        val result = checkResult {
+            taskDao.insertTask(task.toTaskEntity())
+        }
```

Comment:
Since you always end up doing a .fold

You could probably add that to the "checkResult" extension function
So you can just:

```
return checkResult {
            taskDao.insertTask(task.toTaskEntity())
        }
```

Reply:
@MKiperszmid Will update this at a later time, as I would have to change a lot of code.